common_states/states/common_states/initialize_robot/initialize_robot_UYWXTI/script.py

```diff
@@ -0,0 +1,49 @@
+from fetch_api import Base, Arm, Gripper, Torso, Head
+from uw_manipulation import GraspingClient
+import knowledge_representation
+import rospy
+import tf
+import traceback
+
+
+def execute(self, inputs, outputs, gvm):
+    # check if the roscore is already running
+    try:
+        rospy.wait_for_service("/rosout/get_loggers", 5.0)
+    except Exception as e:
+        self.logger.error("Exception: " + str(e) + str(traceback.format_exc()))
+        return -1
+    if "node_name" in inputs:
+        node_name = inputs["node_name"]
+    else:
+        # TODO(nickswalker): Pull this from the parent state machine's name...
+        node_name = "task"
+    if type(node_name).__name__ == "unicode":
+        node_name = node_name.encode('ascii', 'ignore')
+        # print node_name
+
+    self.logger.info("Creating node: " + node_name)
+    try:
+        if not gvm.variable_exist("ros_node_initialized") or not gvm.get_variable("ros_node_initialized"):
+            gvm.set_variable("ros_node_initialized", True)
+            rospy.init_node(node_name, disable_signals=True)
+
+            listener = tf.TransformListener()
+            self.logger.info("Creating node: " + str(listener))
+            gvm.set_variable("tf_listener", listener, per_reference=True)
+            arm = Arm()
+            self.logger.info("tucking arm")
+            arm.tuck()
+            torso = Torso()
+            self.logger.info("lowering torso")
+            torso.set_height(0)
+        
+            fetch = (Base(), arm, Gripper(), GraspingClient(), Head(), torso)
+            gvm.set_variable("robot", fetch, per_reference=True)
+            gvm.set_variable("rospy", rospy, per_reference = True)
```

Comment:
This looks suprising, because other parts of the state machine can import most other modules themselves. I assume rospy has some strange side effect? What does keeping the module around do? Do leave a comment describing why this needs to be here